frontend/src/pages/modelRegistry/screens/ModelVersionDetails/ModelVersionRegisteredFromLink.tsx

```diff
@@ -0,0 +1,174 @@
+      <b>{runName}</b>
+    );
+
+    const renderProject = registeredPipelineProject
+      ? registeredPipelineProject.metadata.annotations?.['openshift.io/display-name'] ??
+        registeredPipelineProject.metadata.name
+      : project;
+
+    return (
+      <Flex
+        spaceItems={{ default: 'spaceItemsXs' }}
+        alignItems={{ default: 'alignItemsCenter' }}
+        data-testid="registered-from-pipeline"
+      >
+        <FlexItem data-testid="pipeline-run-link">Run {renderRunLink} in</FlexItem>
+        <FlexItem style={{ display: 'flex' }}>
+          <TypedObjectIcon
+            resourceType={ProjectObjectType.project}
+            style={{ height: 24, width: 24 }}
+          />
+        </FlexItem>
+        <FlexItem>
+          <b>{renderProject}</b>
+        </FlexItem>
+        {catalogModelUrl && (
+          <FlexItem data-testid="catalog-model-info">
+            <span style={{ fontWeight: 'var(--pf-t--global--font--weight--body--bold)' }}>
+              (from Model Catalog)
+            </span>
+          </FlexItem>
+        )}
+        {!registeredPipelineProject && !catalogModelUrl && (
+          <FlexItem>
+            <Popover
+              data-testid="project-access-info-popover"
+              bodyContent={
+                <Stack hasGutter>
+                  <StackItem>
+                    <Content>
+                      You don&apos;t have access to the source run. To request access, contact your
+                      administrator.
+                    </Content>
+                  </StackItem>
+                  <StackItem>
+                    <Content>Your administrator might be:</Content>
+                    <PopoverListContent
+                      data-testid="pipeline-project-button-popover"
+                      listItems={[...FindAdministratorOptions, 'Your professor (at a school)']}
+                    />
+                  </StackItem>
+                </Stack>
+              }
+            >
+              <Button
+                data-testid="project-access-info-button"
+                icon={<OutlinedQuestionCircleIcon />}
+                variant="link"
+                aria-label="more info"
+              />
+            </Popover>
+          </FlexItem>
+        )}
+      </Flex>
+    );
+  }
+
+  // If we only have catalog details, render the catalog link
+  if (registeredFromCatalogDetails && catalogModelUrl) {
+    return (
+      <a
+        href={catalogModelUrl}
+        data-testid="registered-from-catalog"
+        style={{
+          color: 'var(--pf-global--primary-color--100)',
+          textDecoration: 'underline',
+        }}
+      >
+        <span style={{ fontWeight: 'var(--pf-t--global--font--weight--body--bold)' }}>
+          {modelArtifact.modelSourceName || ''} ({modelArtifact.modelSourceId || ''})
+        </span>{' '}
+        in Model catalog
+      </a>
+    );
+  }
+
+  // Fallback case - shouldn't reach here given the earlier checks
+  return null;
```

Comment:
Once you change the `if (registeredFromCatalogDetails && catalogModelUrl) {` to `if (registeredFromCatalogDetails) {` and define `catalogModelUrl` inside it, I bet you can remove this line and TS will be happy that all cases are satisfied. (If not it's fine to leave it)